Sources/AppStoreConnect/Networking/RetryStrategy.swift

```diff
@@ -0,0 +1,47 @@
+import Foundation
+
+/// Wraps retry logic used by the client when encountering recoverable errors.
+///
+/// This enum is intended to be provided as input to the methods on ``AppStoreConnect``. It only
+/// knows how to handle errors from the App Store Connect and Enterprise Program APIs.
+public enum RetryStrategy: Sendable {
+    /// The default strategy. Avoids retrying under any circumstnace.
+    case never
+    /// The strategy that retries on a fixed interval, in seconds, until a limit is reached.
+    case fixedInterval(_ seconds: UInt64, limit: Int)
+    /// The strategy that retries using an exponential backoff approach until a limit is reached.
+    case exponentialBackoff(_ seconds: UInt64, limit: Int, factor: Int = 2)
+    /// The strategy that retries using a user-defined function. No additional waiting is performed.
+    case custom(_ waiter: @Sendable (any Error) async throws -> Void)
+
+    public func retrying<T>(_ block: @Sendable () async throws -> T) async rethrows -> T {
+        var iteration = 0
+
+        while true {
+            do {
+                return try await block()
+            } catch let error as ResponseError where error.isRecoverable {
+                switch self {
+                case .never:
+                    throw error
+                case .fixedInterval(let interval, let limit):
+                    guard iteration < limit else { throw error }
+
+                    try await Task.sleep(nanoseconds: interval * 1_000_000_000)
```

Comment:
It is time! Let us go `iOS 16` and up. This will allow us to offer the usage of `Duration` in all of our APIs.